### PR TITLE
Exchange rate

### PR DESCRIPTION
**Pull request by Janther**
All tests running and the contribution Bonus logic for after 24 hours contribution.

TODO:
Write the test for this scenario

### CODE REVIEW

**Comment by Janther** [comment]
Writing the tests I found a bug with the amounts, tomorrow I'll continue here, don't merge yet.

**Comment by Janther** [comment]
please proceed

File: contracts/Contribution.sol [hunk 1]
```diff
@@ -140,34 +142,52 @@ contract Contribution is Controlled, TokenController {
...
     }
   }
 
+  function tokensToGenerate(uint256 toFund) constant public initialized returns (uint256) {
+    if (getBlockTimestamp() < startTime + 1 days) {
+      return toFund.mul(exchangeRate());
+    }
+
+    if (notCollectedAmountAfter24Hours == 0) {
+      notCollectedAmountAfter24Hours = weiToCollect();
```
Comment (rstormsf) [inline contracts/Contribution.sol:163]: it has to be recorded only once

Comment (rstormsf) [reply contracts/Contribution.sol:163]: won't work.

File: test/utils.js [hunk 2]
```diff
@@ -54,6 +54,13 @@ export const duration = {
   }
 };
 
-export function latestBlock() {
-  return web3.eth.blockNumber;
+export async function latestBlock() {
+  return new Promise((resolve, reject) => {
```
Comment (rstormsf) [inline test/utils.js:58]: why did you change it back to Promise?

File: contracts/Contribution.sol [hunk 3]
```diff
@@ -225,9 +225,6 @@ contract Contribution is Controlled, TokenController {
     if (getBlockTimestamp() <= startTime + 1 days) {
       require(canPurchase[_th]);
     }
-    if (getBlockTimestamp() >= startTime + 1 days && notCollectedAmountAfter24Hours == 0) {
-      notCollectedAmountAfter24Hours = weiToCollect();
```
Comment (rstormsf) [inline contracts/Contribution.sol:209]: we have to record it and write it as a state variable

File: contracts/Contribution.sol [hunk 4]
```diff
@@ -140,34 +142,52 @@ contract Contribution is Controlled, TokenController {
...
-      } else if(totalWeiCollected > weiTilBonusTwentyExist && totalWeiCollected <= weiTilBonusThirtyExist) {
-        rate = 2200;
-      } else {
-        rate = 2000;
-      }
+      rate = 2000;
     }
   }
 
+  function tokensToGenerate(uint256 toFund) constant public initialized returns (uint256) {
```
Comment (rstormsf) [inline contracts/Contribution.sol]: remove constant

**Review by rstormsf — APPROVED** [review]
